### DISCUSSION TITLE
Problem on create collection using API

### DISCUSSION THREAD

**Discussion by AbnerDigital** [pocketbase/pocketbase#6204] [Q&A]
Hello friends, I'm trying to create a collection in pocketdb directly in the code, importing the JSON within the pocketDB administrative panel it imports the collection correctly, but if I try to make the application create it using the API it creates the collection correctly but it can't create any collection field, follow the code I'm using.

`import { pb } from './pocketbase';

const createCollections = async () => {
    console.log('Iniciando criação das collections...');
    
    try {
        const collectionData = [
            {
                name: "transactions",
                type: "base",
                schema: [
                    {
                        name: "valor",
                        type: "number",
                        required: true
                    },
                    {
                        name: "descricao",
                        type: "text",
                        required: true
                    },
                    {
                        name: "dataRecebimento",
                        type: "date",
                        required: true
                    },
                    {
                        name: "company",
                        type: "relation",
                        required: false,
                        options: {
                            maxSelect: 1,
                            collectionId: "companies",
                            cascadeDelete: false
                        }
                    }
                ]
            },
            {
                name: "companies",
                type: "base",
                schema: [
                    {
                        name: "nome",
                        type: "text",
                        required: true
                    }
                ]
            }
        ];

        for (const data of collectionData) {
            const existingCollection = await pb.collections.getList(1, 1, {
                filter: `name = "${data.name}"`,
            });

            if (existingCollection.totalItems > 0) {
                console.log(`A collection "${data.name}" já existe. Pulando a criação.`);
                continue;
            }

            console.log(`Tentando criar a collection "${data.name}" com a estrutura:`, JSON.stringify(data, null, 2));

            // Criando a collection com os campos personalizados
            const newCollection = await pb.collections.create(data);
            console.log(`Collection "${data.name}" criada com sucesso:`, JSON.stringify(newCollection, null, 2));

            // Verificando a collection criada
            const verifyCollection = await pb.collections.getOne(newCollection.id);
            console.log(`Verificação da collection "${data.name}" criada:`, JSON.stringify(verifyCollection, null, 2));
        }

    } catch (error) {
        console.error('Erro ao criar collections:', error.message);
        console.error('Detalhes do erro:', error);
    }
};

export default createCollections;

[import.json](https://github.com/user-attachments/files/18272387/import.json)
`
I attached the JSON schema that I'm using as a base exported directly from pocketdb, as I can't create it directly in the API, I created it manually to export the model and analyze it and I still can't, i'm use latest release on 30/12/2024 from pocketdb.

**Comment by ganigeorgiev** [comment]
**If you used LLM to generate the above, please stop doing it!**

In any case, it is not clear what you are trying to do, but the above collections data is incorrect - there is no such field as "schema" or "options" in the most recent releases. See the examples in https://pocketbase.io/docs/api-collections/#create-collection (**all collection operations require superuser authentication**).

It is also not clear what do you refer to by "30/12/2024 from pocketdb". There is no such thing.

If you need further help, please elaborate more on your use case but don't use llm.
